hal_st/instantiations/NucleoUi.hpp

```diff
@@ -17,6 +19,15 @@ namespace main_
 #else
         hal::GpioPinStm ledGreen{ hal::Port::B, 3 };
 #endif
+
+#if defined(STM32G0) || defined(STM32G4)
+        hal::GpioPinStm stLinkUartTxPin{ hal::Port::A, 2 };
+        hal::GpioPinStm stLinkUartRxPin{ hal::Port::A, 3 };
+
+        hal::DmaStm dmaStm;
+        hal::DmaStm::TransmitStream transmitStream{ dmaStm, hal::DmaChannelId{ 1, 1, DMA_REQUEST_USART2_TX } };
+        hal::UartStmDma stLinkUartDma{ transmitStream, 2, stLinkUartTxPin, stLinkUartRxPin };
```

Comment:
These UI classes should not contain DMA or UART definitions. They should only contain what you could optionally use. Note that there is a GpioPinStm for the LEDs, but no corresponding OutputPin to actually activate them.